examples/externalContactsDemo/script.js

```diff
@@ -0,0 +1,269 @@
+    client.setEnvironment(pcEnvironment);
+
+    let clientApp = null;
+    try {
+        clientApp = new window.purecloud.apps.ClientApp({
+            pcEnvironment: pcEnvironment
+        });
+    } catch (e) {
+        setErrorState(
+            pcEnvironment + ': Unknown/Unsupported PureCloud Embed Context'
+        );
+        return;
+    }
+    updateProgressBar(80);
+
+    /*
+    *  Set up manual tester now; no PC Auth required
+    *
+    *  This UI is used for testing purposes to manually navigate to an external contact by ID.
+    *  It purposefully ignores the fact that a user may not be authorized to view external contacts.
+    */
+    document
+        .querySelector('.manual-example #contactButton')
+        .addEventListener('click', function () {
+            let contactId = document.querySelector('#contact-id').value.trim();
+            if (contactId) {
+                clientApp.externalContacts.showExternalContactProfile(contactId);
+            }
+        });
+    document
+        .querySelector('.manual-example #orgButton')
+        .addEventListener('click', function () {
+            let externalOrganizationId = document.querySelector('#org-id').value.trim();
+            if (externalOrganizationId) {
+                clientApp.externalContacts.showExternalOrganizationProfile(externalOrganizationId);
+            }
+        });
+
+    let redirectUrl = window.location.origin;
+    if (!redirectUrl) {
+        redirectUrl = window.location.protocol + '//' + window.location.host;
+    }
+    redirectUrl += window.location.pathname;
+
+    // Authenticate with PureCloud
+    let authenticated = false;
+    let userDataAcquired = false;
+
+    client
+        .loginImplicitGrant(clientId, redirectUrl, {
+            state: 'pcEnvironment=' + pcEnvironment
+        })
+        .then(function () {
+            updateProgressBar(100);
+            authenticated = true;
+            return new platformClient.UsersApi().getUsersMe({
+                expand: ['authorization']
+            });
+        })
+        .then(function (profileData) {
+            userDataAcquired = true;
+            // Check if the current user will be able to view external contacts
+            let permissions = profileData.authorization.permissions;
+            if (checkPermission(permissions, 'externalContacts:contact:view')) {
+                requestExternalContacts().then(data => {
+                    let entities = data.entities;
+                    // Build table with data from api call
+                    for (let i = 0; i < entities.length; i++) {
+                        let hasOrg = !!entities[i].externalOrganization;
+                        document.getElementById('tableBody').insertAdjacentHTML('beforeend',
+                            `<tr>
+                                <td>
+                                    <a id="entity-${i}">${entities[i].lastName}, ${entities[i].firstName}</a>
```

Comment:
I wonder if you could create the nodes first and then come back through with innerText to insert this user data so it can't be XSS?  Sorry, I know this is a PITA and just a demo app, but prob worth doing.